test/server/connection_handler_test.cc

```diff
@@ -1546,6 +1575,68 @@ TEST_F(ConnectionHandlerTest, ShutdownUdpListener) {
       << "The read_filter_ should be deleted before the udp_listener_ is deleted.";
 }
 
+TEST_F(ConnectionHandlerTest, DisableInternalListener) {
+  InSequence s;
+  Network::Address::InstanceConstSharedPtr local_address{
+      new Network::Address::EnvoyInternalInstance("server_internal_address")};
+
+  TestListener* internal_listener =
+      addInternalListener(1, "test_internal_listener", std::chrono::milliseconds(), false, nullptr);
+  EXPECT_CALL(internal_listener->socket_factory_, localAddress())
+      .WillOnce(ReturnRef(local_address));
+  handler_->addListener(absl::nullopt, *internal_listener);
+  auto internal_listener_cb = handler_->findByAddress(local_address);
+  ASSERT(internal_listener_cb.has_value());
```

Comment:
Better use gtest's assertions here and in other places
```suggestion
  ASSERT_TRUE(internal_listener_cb.has_value());
```

Reply:
TIL. Many thanks!